File: qc-domain/src/main/java/com/eager/questioncloud/domain/cart/repository/CartItemRepository.java
package com.eager.questioncloud.domain.cart.repository;

import com.eager.questioncloud.domain.cart.dto.CartItemDetail;
import com.eager.questioncloud.domain.cart.model.CartItem;
import java.util.List;

public interface CartItemRepository {
    CartItem save(CartItem cartItem);

    List<CartItemDetail> findByUserId(Long userId);

    void deleteByIdInAndUserId(List<Long> ids, Long userId);

    void deleteByQuestionIdInAndUserId(List<Long> questionIds, Long userId);

    Boolean isExistsInCart(Long userId, Long questionId);
}

File: qc-domain/qc-cart/src/main/kotlin/com/eager/questioncloud/cart/infrastructure/repository/CartItemRepository.kt
package com.eager.questioncloud.cart.infrastructure.repository

import com.eager.questioncloud.cart.domain.CartItem

interface CartItemRepository {
    fun save(cartItem: CartItem): CartItem

    fun findByUserId(userId: Long): List<CartItem>

    fun deleteByIdInAndUserId(ids: List<Long>, userId: Long)

    fun deleteByQuestionIdInAndUserId(questionIds: List<Long>, userId: Long)

    fun isExistsInCart(userId: Long, questionId: Long): Boolean

    fun deleteAllInBatch()
}
